File: backend/src/Modules/Social/Dtos/FriendshipDto.cs
namespace Trippie.Modules.Social.Dtos;

public sealed class FriendshipDto
{
	public required int Id { get; set; }
	public required int FriendId { get; set; }
	public required DateTime CreatedAt { get; set; }
}

public sealed class CreateFriendshipDto
{
	public required int FriendId { get; set; }
}

public sealed class FriendDto
{
	public required int FriendId { get; set; }
	public required string Username { get; set; }
    public string? ProfilePhotoUrl { get; set; }
}

File: frontend/src/app/core/feature/dtos/social/friendship.dto.ts
import { UserDto } from '../../auth/dtos/user.dto';

export interface FriendshipDto {
  id: number;
  userId1: number;
  user1: UserDto;
  userId2: number;
  user2: UserDto;
  createdAt: string;
}

export interface CreateFriendshipDto {
  userId1: number;
  userId2: number;
}

export interface FriendDto {
  id: number;
  friendId: number;
  username: string;
  profilePhotoUrl: string | null;
}
